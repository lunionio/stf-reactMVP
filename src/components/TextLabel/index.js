import React from "react";
import styled from "styled-components/native";

const Label = styled.Text`
  ${props =>
    props.color &&
    `
      color: ${props.color};
  `};

  ${props =>
    props.fontFamily &&
    `
      ${props.fontFamily};
  `};

  ${props =>
    props.fontWeight &&
    `
      font-weight: ${props.fontWeight};
  `};

  ${props =>
    props.fontSize &&
    `
      font-size: ${props.fontSize};
  `};

  ${props =>
    props.textAlign &&
    `
      text-align: ${props.textAlign}
  `};
`;

const TextLabel = ({
  color,
  fontFamily,
  fontWeight,
  fontSize,
  label,
  textAlign
}) => {
  return (
    <Label
      color={color}
      fontWeight={fontWeight}
      fontFamily={fontFamily}
      fontSize={fontSize}
      textAlign={textAlign}
    >
      {label}
    </Label>
  );
};

export default TextLabel;
