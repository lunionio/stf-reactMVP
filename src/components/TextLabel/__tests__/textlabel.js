import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import TextLabel from '../index'

describe('Components - TextLabel', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TextLabel />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
