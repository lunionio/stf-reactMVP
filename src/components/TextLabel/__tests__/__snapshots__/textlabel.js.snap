// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - TextLabel renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {},
      undefined,
    ]
  }
/>
`;
